### ISSUE TITLE
Bug: Error: client must use upgraded connection to attach

### ISSUE THREAD

**Issue by cmoulliard** [boot2podman/machine#16]
## Issue

When we execute the following `podman-remote` command to run a busybox container, we get this error
```
podman-remote run busybox echo hello world
ERRO[0000] unable to make temp dir /run/user/-1
Error: client must use upgraded connection to attach
```

## Info
```
podman-remote -v
podman-remote version 1.4.5-dev
```

**Comment by afbjorklund** [comment]
I saw this myself, but I have no idea what error is or how to fix ? Maybe upgrading varlink would help ?

Anyway, this seems like upstream issue. It works fine over regular ssh.

**Comment by afbjorklund** [comment]
Upgrading didn't help, unfortunately.

`Error: client must use upgraded connection to attach`

Seems to be an upstream limitation?
```console
tc@box:~$ varlink --version
18
```

----

```console
$ podman-remote version
Client:
Version:            1.6.0-dev
RemoteAPI Version:  1
Go Version:         go1.12.10
OS/Arch:            linux/amd64

Service:
Version:            1.6.0-dev
RemoteAPI Version:  1
Go Version:         go1.12.10
OS/Arch:            linux/amd64

```

**Comment by cmoulliard** [comment]
We still get the same error even using `podman-version` version 1.6.0
```
podman-remote -v
podman-remote version 1.6.0-dev
podman-remote run busybox echo hello world
Error: client must use upgraded connection to attach
```

**Comment by afbjorklund** [comment]
I opened up https://github.com/containers/libpod/issues/4138 with some more details.

Basically I think it *only* works over ssh (not bridge), and it seems like the syntax is wrong on Windows (but haven’t tested that part myself)

**Comment by cmoulliard** [comment]
As the ticket is closed on containers/libpod, what do you then suggest to do for mac users to resolve this issue ?

**Comment by afbjorklund** [comment]
As far as I know, the issue still remains (upstream). I fixed the variables and stuff, but haven’t revisited the main issue here...

The workaround is to use ssh, rather than try to use interactive varlink.

**Comment by afbjorklund** [comment]
Tested with the new 0.25 release (podman 1.7.1-dev)

There is some difference in how varlink is being called.


This works (`podman-machine env`):

```
  Connection: ssh -p 40164 -T -i /home/anders/.local/machine/machines/box/id_rsa -q
    -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null root@127.0.0.1 --
    varlink -A \'podman --log-level=debug varlink \\\$VARLINK_ADDRESS\' bridge
```

```console
$ podman-remote run -it busybox
/ #
```

This does **not** (`podman-machine env --varlink`)

```
  Connection: /usr/bin/ssh -F /dev/null -o ConnectionAttempts=3 -o ConnectTimeout=10
    -o ControlMaster=no -o ControlPath=none -o LogLevel=quiet -o PasswordAuthentication=no
    -o ServerAliveInterval=60 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
    tc@127.0.0.1 -o IdentitiesOnly=yes -i /home/anders/.local/machine/machines/box/id_rsa
    -p 40164 sudo varlink bridge
```

```console
$ podman-remote run -it busybox
Error: client must use upgraded connection to attach
```

Maybe we can pass that -A parameter ?

**Comment by afbjorklund** [comment]
Yes, adding the -A parameter did the trick.

Fixed in 23fdcc2d4905193c1e31807f783087c0918c7517

**Comment by afbjorklund** [comment]
Closed in `v0.17`
